src/modele/Customer.java

```diff
@@ -40,58 +34,81 @@
 @DiscriminatorValue("2")
 public class Customer extends Point implements Serializable {
 
-    private static final long serialVersionUID = 1L;
-    @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    private Long id;
     @Basic(optional = false)
     @Column(name = "POSITION")
     private Integer position;
+
     @OneToMany(cascade = CascadeType.ALL, mappedBy = "customer")
     private Set<Demand> customerDemands;
-
+    
+    /**
+     * No-argument constructor
+     */
     public Customer() {
         super();
         this.position = 0;
-        this.customerDemands = new HashSet<Demand>();
+        this.customerDemands = new HashSet<>();
     }
 
-    public Customer(double x, double y, int demand) {
-        super(2, x, y);
+    /**
+     * Parameterized constructor
+     *
+     * @param id
+     * @param x
+     * @param y
+     * @param demand
+     */
+    public Customer(Integer id, double x, double y, int demand) {
```

Comment:
mais l'id est généré automùatiquement, non ?

Reply:
id are given in the text files
it will be best to follow them, because locations are associated with ids